### ISSUE TITLE
some wacky align-horizontal buisness going on

### ISSUE THREAD

**Issue by TryMyWagon** [Textualize/textual#5345]
for some reason using "align" for me in css is affecting the class and not just the children of the class, which was leading to a funky interaction with "dock", i think its trying to align itself to the center of the docked position, eg

having "dock: right;" and "align-horizontal: center;" in a widgets css class would result in the widget ending up aligning itself relative to the center of the dock

in the likely event im a dumbass and missed some obvious important documentation please forgive me for taking up your time
im using the funky windows terminal with powershell in a venv with recent pip installs of rich and textual

video of the wacky stuff:
https://github.com/user-attachments/assets/584f5857-485c-4a7c-bf10-1d37dc56683d

the code causing said wack stuff:
[test.zip](https://github.com/user-attachments/files/18022898/test.zip)

ps this library is amazing and "Rodrigo here for textualise" is a better teacher than all of my profs combined :D tysm!

```
# Textual Diagnostics


## Versions

| Name    | Value  |
|---------|--------|
| Textual | 0.88.1 |
| Rich    | 13.9.4 |

## Python

| Name           | Value                                                                                                            |
|----------------|------------------------------------------------------------------------------------------------------------------|
| Version        | 3.12.1                                                                                                           |
| Implementation | CPython                                                                                                          |
| Compiler       | MSC v.1937 64 bit (AMD64)                                                                                        |
| Executable     | C:\Users\Miles\Documents\!Programs(LOCAL)\!Projects\(python) TwitchOW2PredictionBot\src\.venv\Scripts\python.exe |

## Operating System

| Name    | Value      |
|---------|------------|
| System  | Windows    |
| Release | 10         |
| Version | 10.0.19045 |

## Terminal

| Name                 | Value     |
|----------------------|-----------|
| Terminal Application | *Unknown* | 
| TERM                 | *Not set* |
| COLORTERM            | *Not set* |
| FORCE_COLOR          | *Not set* |
| NO_COLOR             | *Not set* |

## Rich Console options

| Name           | Value                |
|----------------|----------------------|
| size           | width=120, height=30 |
| legacy_windows | False                |
| min_width      | 1                    |
| max_width      | 120                  |
| is_terminal    | True                 |
| encoding       | utf-8                |
| max_height     | 30                   |
| justify        | None                 |
| overflow       | None                 |
| no_wrap        | False                |
| highlight      | None                 |
| markup         | None                 |
| height         | None                 |
```

**Comment by willmcgugan** [comment]
@rodrigogiraoserrao continues to receive praise for his teaching style...

**Comment by willmcgugan** [comment]
Fixed in v0.89.1

**Comment by rodrigogiraoserrao** [comment]
Thanks for the tag, @willmcgugan, and thanks for the kind words, @TryMyWagon!
